libsoftwarecontainer/src/gateway/dbus/dbusgatewayparser.cpp

```diff
@@ -19,45 +19,25 @@
 
 #include "dbusgatewayparser.h"
 
-
-ReturnCode DBusGatewayParser::parseDBusConfigElement(const json_t *element,
-                                                     json_t *session,
-                                                     json_t *system)
+ReturnCode DBusGatewayParser::parseDBusConfig(const json_t *element,
+                                              const char *key,
+                                              json_t *config)
 {
-
-    if (!JSONParser::hasKey(element, SESSION_CONFIG)
-     && !JSONParser::hasKey(element, SYSTEM_CONFIG)) {
-        log_error() << "Neither system nor session configuration was provided";
-        return ReturnCode::FAILURE;
-    }
-
-    if (isError(parseDBusBusConfig(element, SESSION_CONFIG, session))) {
+    json_t *configExists = json_object_get(element, key);
+    if (configExists == nullptr) {
```

Comment:
The name `configExists` is not very nice either, use `config` or something similar instead.